### PR TITLE
Add Status for Assistant Tasks

### PR DESCRIPTION
**Pull request by alperozturk96**
- [x] Tests written, or not not needed

Screenshot 2024-03-27 at 16 23 49

**How to Test**

Add Nextcloud Assistant from Web UI

Relaunch app

Create task

### CODE REVIEW

File: app/src/main/java/com/nextcloud/client/assistant/extensions/TaskExtensions.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/*
+ * Nextcloud Android Library
+ *
+ * SPDX-FileCopyrightText: 2023-2024 Nextcloud GmbH and Nextcloud
+ * contributors
+ * SPDX-FileCopyrightText: 2023 Alper Ozturk <alper_ozturk@proton.me>
+ * SPDX-License-Identifier: MIT
+ */
+
```
Comment (tobiasKaminsky) [inline app/src/main/java/com/nextcloud/client/assistant/extensions/TaskExtensions.kt:9]: Wrong license.

File: app/src/main/java/com/owncloud/android/MainApp.java [hunk 2]
```diff
@@ -674,32 +681,32 @@ public static String getAccountType(Context context) {
     //  From AccountAuthenticator
     //  public static final String AUTHORITY = "org.owncloud";
     public static String getAuthority() {
-        return getAppContext().getResources().getString(R.string.authority);
+        return string(R.string.authority);
```
Comment (tobiasKaminsky) [inline app/src/main/java/com/owncloud/android/MainApp.java:684]: please do not refactor unrelated methods -> new PR

File: app/src/main/java/com/nextcloud/client/assistant/AssistantViewModel.kt [hunk 3]
```diff
@@ -91,12 +91,14 @@ class AssistantViewModel(private val repository: AssistantRepositoryType) : View
 
     private fun getTaskTypes() {
         viewModelScope.launch(Dispatchers.IO) {
-            val allTaskType = MainApp.getAppContext().getString(R.string.assistant_screen_all_task_type)
+            val allTaskType = MainApp.string(R.string.assistant_screen_all_task_type)
```
Comment (tobiasKaminsky) [inline app/src/main/java/com/nextcloud/client/assistant/AssistantViewModel.kt:94]: Isn't there another way? Pass this string as constructor?
Having a dependecy to MainApp should be avoided.

**Review by tobiasKaminsky — DISMISSED** [review]
Please remove dependency to MainApp

File: app/src/main/java/com/nextcloud/client/assistant/extensions/TaskExtensions.kt (hunk 1 above)
Comment (alperozturk96) [reply app/src/main/java/com/nextcloud/client/assistant/extensions/TaskExtensions.kt:9]: What's the correct one?

**Review by JonasMayerDev — APPROVED** [review]
New feature works in my testing

**Comment by alperozturk96** [comment]
Duplication [of](https://github.com/nextcloud/android/pull/12791)

**Pull request closed without merging**